Tests/SwiftGRPCNIOTests/NIOServerTests.swift

```diff
@@ -92,211 +94,106 @@ class NIOServerTests: NIOServerTestCase {
-    XCTAssertEqual("Swift echo expand (1): bar", try! call.receive()!.text)
-    XCTAssertEqual("Swift echo expand (2): baz", try! call.receive()!.text)
-    XCTAssertNil(try! call.receive())
+    XCTAssertEqual(try call.status.wait().code, .ok)
+  }
 
-    waitForExpectations(timeout: defaultTimeout)
+  func testServerStreaming() {
+    XCTAssertNoThrow(try doTestServerStreaming(messages: NIOServerTests.aFewStrings))
   }
 
   func testServerStreamingLotsOfMessages() {
-    let completionHandlerExpectation = expectation(description: "completion handler called")
-    let call = try! client.expand(Echo_EchoRequest(text: NIOServerTests.lotsOfStrings.joined(separator: " "))) { callResult in
-      XCTAssertEqual(.ok, callResult.statusCode)
-      completionHandlerExpectation.fulfill()
-    }
-
-    for string in NIOServerTests.lotsOfStrings {
-      XCTAssertEqual("Swift echo expand (\(string)): \(string)", try! call.receive()!.text)
-    }
-    XCTAssertNil(try! call.receive())
-
-    waitForExpectations(timeout: defaultTimeout)
+    XCTAssertNoThrow(try doTestServerStreaming(messages: NIOServerTests.lotsOfStrings))
   }
 }
 
 extension NIOServerTests {
-  func testBidirectionalStreamingBatched() {
-    let finalCompletionHandlerExpectation = expectation(description: "final completion handler called")
-    let call = try! client.update { callResult in
-      XCTAssertEqual(.ok, callResult.statusCode)
-      finalCompletionHandlerExpectation.fulfill()
+  private func doTestBidirectionalStreaming(messages: [String], waitForEachResponse: Bool = false) throws {
+    let responseReceived = waitForEachResponse ? DispatchSemaphore(value: 0) : nil
+    var index = 0
+
+    let call = client.update { response in
+      XCTAssertEqual("Swift echo update (\(index)): \(messages[index])", response.text)
+      responseReceived?.signal()
+      index += 1
     }
 
-    XCTAssertNoThrow(try call.send(Echo_EchoRequest(text: "foo")))
-    XCTAssertNoThrow(try call.send(Echo_EchoRequest(text: "bar")))
-    XCTAssertNoThrow(try call.send(Echo_EchoRequest(text: "baz")))
-
-    call.waitForSendOperationsToFinish()
-
-    XCTAssertNoThrow(try call.closeSend())
-
-    XCTAssertEqual("Swift echo update (0): foo", try! call.receive()!.text)
-    XCTAssertEqual("Swift echo update (1): bar", try! call.receive()!.text)
-    XCTAssertEqual("Swift echo update (2): baz", try! call.receive()!.text)
-    XCTAssertNil(try! call.receive())
-
-    waitForExpectations(timeout: defaultTimeout)
-  }
-
-  func testBidirectionalStreamingPingPong() {
-    let finalCompletionHandlerExpectation = expectation(description: "final completion handler called")
-    let call = try! client.update { callResult in
-      XCTAssertEqual(.ok, callResult.statusCode)
-      finalCompletionHandlerExpectation.fulfill()
+    messages.forEach { part in
+      call.send(.message(Echo_EchoRequest.with { $0.text = part }))
+      responseReceived?.wait()
```

Comment:
I'd add a timeout to the `wait` (again, to prevent stalling tests) and assert that the result of `wait` is equal to (not `timeoutExceeded`, I've forgotten what the other enum case is called)